crates/api/src/fetch.rs

```diff
@@ -1,59 +1,72 @@
 //! Kitsune2 fetch types.
 
-use std::sync::Arc;
-
+use bytes::BytesMut;
+use k2_fetch_message::FetchMessageType;
 use prost::Message;
 
 use crate::{
     builder, config, peer_store::DynPeerStore, transport::DynTransport,
-    AgentId, BoxFut, DynOpStore, K2Error, K2Result, MetaOp, OpId, SpaceId,
+    AgentId, BoxFut, DynOpStore, K2Result, MetaOp, OpId, SpaceId,
 };
+use std::sync::Arc;
 
 include!("../proto/gen/kitsune2.fetch.rs");
 
-impl From<Vec<OpId>> for OpIds {
+impl From<Vec<OpId>> for Request {
     fn from(value: Vec<OpId>) -> Self {
         Self {
-            data: value.into_iter().map(Into::into).collect(),
+            op_ids: value.into_iter().map(Into::into).collect(),
         }
     }
 }
 
-impl From<OpIds> for Vec<OpId> {
-    fn from(value: OpIds) -> Self {
-        value.data.into_iter().map(Into::into).collect()
+impl From<Request> for Vec<OpId> {
+    fn from(value: Request) -> Self {
+        value.op_ids.into_iter().map(Into::into).collect()
     }
 }
 
-impl From<Vec<MetaOp>> for Ops {
+impl From<Vec<MetaOp>> for Response {
     fn from(value: Vec<MetaOp>) -> Self {
         Self {
-            op_list: value.into_iter().map(Into::into).collect(),
+            ops: value.into_iter().map(Into::into).collect(),
         }
     }
 }
 
 /// Serialize list of op ids for sending over the wire.
-pub fn serialize_op_ids(value: Vec<OpId>) -> bytes::Bytes {
-    let mut out = bytes::BytesMut::new();
-    OpIds::from(value)
+pub fn serialize_request(value: Vec<OpId>) -> bytes::Bytes {
+    let mut data = BytesMut::new();
```

Comment:
I'd be tempted to do the inner serialize in a separate function. It takes a bit of thought to decode the double encode that's happening in here

Reply:
Cool, done.